pkgs/os-specific/linux/zfs/default.nix

```diff
@@ -170,28 +173,38 @@ let
         maintainers = with maintainers; [ jcumming wizeman fpletz globin ];
       };
     };
-in {
+in mapAttrs (_: common) rec {
   # also check if kernel version constraints in
   # ./nixos/modules/tasks/filesystems/zfs.nix needs
   # to be adapted
-  zfsStable = common {
+  zfsStable = {
     # comment/uncomment if breaking kernel versions are known
     # incompatibleKernelVersion = "4.20";
 
     # this package should point to the latest release.
     version = "0.8.3";
 
     sha256 = "0viql8rnqr32diapkpdsrwm6xj8vw5vi4dk2x2m7s7g0q2zdkahw";
+
+    extraPatches = optionals (buildKernel && versionAtLeast kernel.version "5.6") [
+      # $ git checkout zfs-0.8.3
+      # $ git log --oneline --grep='Linux 5.6 compat' master
+      # (...cherry-pick, fix conflicts...)
+      # $ git format-patch -o ~/src/nixpkgs/pkgs/os-specific/linux/zfs zfs-0.8.3
+      ./0001-Linux-5.6-compat-struct-proc_ops.patch
+      ./0002-Linux-5.6-compat-timestamp_truncate.patch
+      ./0003-Linux-5.6-compat-ktime_get_raw_ts64.patch
+      ./0004-Linux-5.6-compat-time_t.patch
```

Comment:
Any chance any of these can be pulled using `fetchpatch`?

Reply:
Unfortunately, I don't think so. The conflicts were largely due to a mass rename, so git history is required to resolve them.

They could perhaps be derivations that are functions of those `fetchpatch`s plus the entire OpenZFS git tree, with some magic to ensure the m4 conflict is resolved automatically, or have custom logic to rewrite the paths in the patches, but I'm not sure the auditability benefit over comparing diffs would be worth it.

Reply:
@emilazy This is fine then.